tests/foreman/destructive/test_katello_agent.py

```diff
@@ -47,63 +47,22 @@ def sat_with_katello_agent(module_target_sat):
     yield module_target_sat
 
 
-@pytest.fixture(scope='module')
-def katello_agent_repos(sat_with_katello_agent):
-    """Create Org, Lifecycle Environment, Content View, Activation key"""
-    sat = sat_with_katello_agent
-    org = sat.api.Organization().create()
-    lce = sat.api.LifecycleEnvironment(organization=org).create()
-    cv = sat.api.ContentView(organization=org).create()
-    ak = sat.api.ActivationKey(environment=lce, organization=org).create()
-    sat.cli_factory.setup_org_for_a_rh_repo(
-        {
-            'product': constants.PRDS['rhel'],
-            'repository-set': constants.REPOSET['rhst7'],
-            'repository': constants.REPOS['rhst7']['name'],
-            'organization-id': org.id,
-            'content-view-id': cv.id,
-            'lifecycle-environment-id': lce.id,
-            'activationkey-id': ak.id,
-        }
-    )
-    # Create custom repository content
-    sat.cli_factory.setup_org_for_a_custom_repo(
-        {
-            'url': settings.repos.yum_1.url,
-            'organization-id': org.id,
-            'content-view-id': cv.id,
-            'lifecycle-environment-id': lce.id,
-            'activationkey-id': ak.id,
-        }
-    )
-    return {
-        'ak': ak,
-        'cv': cv,
-        'lce': lce,
-        'org': org,
-    }, sat_with_katello_agent
-
-
 @pytest.fixture
-def katello_agent_client(katello_agent_repos, rhel7_contenthost):
-    katello_agent_repos, sat_with_katello_agent = katello_agent_repos
-    rhel7_contenthost.install_katello_ca(sat_with_katello_agent)
-    # Register content host and install katello-agent
-    rhel7_contenthost.register_contenthost(
-        katello_agent_repos['org'].label,
-        katello_agent_repos['ak'].name,
+def katello_agent_client(sat_with_katello_agent, rhel_contenthost):
+    """Register content host to Satellite and install katello-agent on the host."""
+    org = sat_with_katello_agent.api.Organization().create()
+    client_repo = settings.repos['SATCLIENT_REPO'][f'RHEL{rhel_contenthost.os_version.major}']
+    sat_with_katello_agent.register_host_custom_repo(
+        org, rhel_contenthost, [client_repo, settings.repos.yum_1.url]
     )
-    assert rhel7_contenthost.subscribed
-    host_info = sat_with_katello_agent.cli.Host.info({'name': rhel7_contenthost.hostname})
-    rhel7_contenthost.enable_repo(constants.REPOS['rhst7']['id'])
-    rhel7_contenthost.install_katello_agent()
-    yield {'client': rhel7_contenthost, 'host_info': host_info, 'sat': sat_with_katello_agent}
+    rhel_contenthost.install_katello_agent()
+    host_info = sat_with_katello_agent.cli.Host.info({'name': rhel_contenthost.hostname})
+    yield {'client': rhel_contenthost, 'host_info': host_info, 'sat': sat_with_katello_agent}
 
 
-@pytest.mark.tier3
-@pytest.mark.upgrade
+@pytest.mark.rhel_ver_match(r'[\d]+')
```

Comment:
@rplevka Check me here, but I'm pretty sure this is redundant, the default will be to parametrize against available rhel versions, and this just matches everything.

Reply:
@ogajduse I see your comment above - but you suggested a regex that ends in `$`, excluding anything that doesn't end in a digit. This expression won't exclude `_something` contenthost types/verisons.

Reply:
@mshriver When I tried this before pushing, it collected all RHEL versions except `_fips`